src/pages/rayon.md

```diff
@@ -0,0 +1,46 @@
+# Rayon
+[![rayon-badge]][rayon]
+
+## Calculate difference of value 1 from each element of an array
+
+```rust
+
+/**
+ * An example of parallel processing by using rayon where difference of
+ * 1 is calculated on each element of array in parallel
+ */
+
+extern crate rayon;
+use rayon::prelude::*;
+
+//function to calculate difference of 1 one on each element of an array
+fn difference(input: &mut[i32]){
+    // par_iter_mut() uses internal thread pool to easily parallelize data-parallel operations
+    input.par_iter_mut().for_each(|p| *p -= 1);
+    //print the output,i.e.the new mutated elements of array
+    println!("The mutated array elements are:");
+    for i in input.iter() {
+        println!("{}", *i)
+    };
+}
+
+fn main() {
+    let arr = &mut [0, 7, 9, 11];
+    difference(arr);
+}
+```
+
+The purpose of this is to display the raw power of the rayon library. Rayon is a data-parallelism library for Rust. Ifunction in the example abovet is extremely lightweight and makes it easy to convert a sequential computation into a parallel one. It also guarantees data-race freedom. It also supports an API called "parallel iterators"; we can see the
+par_iter_mut() in the example above . These let you write iterator-like chains that execute in parallel.
+
+The first step involves defining a function 'difference' which calculates a difference of 1 from each element of array.
+We have given a reference to a mutable array of type i32 as an argument to the function. Then, the parallel iterator,
+par_iter_mut(),computes the difference of 1 from each element in parallel rather than sequentially.
+
+Finally we print all elements of the array to check if our code is producing correct results.
```

Comment:
Let's consider who the audience is, and what they are here for. They know Rust at this point, and know they need a library for some purpose. They want to know how to iterate in parallel. Let's give them what they want as clearly as we can without a lot of ceremony.

So the last sentence about printing the array, our user doesn't need to be told that - they know how to print arrays already. Likewise, there's a lot of description here of what the function is doing. How much does the user care about that? They'll recognize that it's a function that subtracts 1.

Basically, the information to convey here is that this example uses the `rayon` crate, which defines a trait (what trait is it), that provides a `par_iter_mut` method. What does that method do? What about that `for_each` method?

Please consider these questions and recompose the description of this example.